tt_metal/impl/device/device_pool.cpp

```diff
@@ -443,19 +449,69 @@ void DevicePool::add_devices_to_pool(const std::vector<chip_id_t>& device_ids) {
-    if (fabric_config == FabricConfig::FABRIC_2D || fabric_config == FabricConfig::FABRIC_2D_PUSH) {
+    if (fabric_config == FabricConfig::FABRIC_1D) {
+        using namespace tt::tt_fabric;
+        static constexpr std::size_t edm_buffer_size =
+            tt::tt_fabric::FabricEriscDatamoverBuilder::default_packet_payload_size_bytes +
+            sizeof(tt::tt_fabric::PacketHeader);
+        const auto edm_config = tt::tt_fabric::FabricEriscDatamoverConfig(edm_buffer_size, 1, 2);
+        auto routing_directions = {RoutingDirection::N, RoutingDirection::S, RoutingDirection::E, RoutingDirection::W};
+        std::vector<uint32_t> signal(1, tt::tt_fabric::EDMStatus::READY_FOR_TRAFFIC);
+
+        auto wait_for_handshake = [&](IDevice* dev) {
+            auto [mesh_id, chip_id] = this->control_plane->get_mesh_chip_id_from_physical_chip_id(dev->id());
+            for (const auto& direction : routing_directions) {
+                auto fabric_active_eth_chans =
+                    control_plane->get_active_fabric_eth_channels_in_direction(mesh_id, chip_id, direction);
+                if (fabric_active_eth_chans.empty()) {
+                    continue;
+                }
+                auto neighbors = this->control_plane->get_intra_chip_neighbors(mesh_id, chip_id, direction);
+                if (neighbors.empty()) {
+                    continue;
+                }
+                for (const auto& eth_chan : fabric_active_eth_chans) {
+                    std::vector<std::uint32_t> router_status{0};
+                    auto eth_logical_core = tt::Cluster::instance().get_soc_desc(dev->id()).get_eth_core_for_channel(
+                        eth_chan, CoordSystem::LOGICAL);
+                    while (router_status[0] != tt::tt_fabric::EDMStatus::HANDSHAKE_COMPLETE) {
+                        tt_metal::detail::ReadFromDeviceL1(
+                            dev, eth_logical_core, edm_config.edm_status_address, 4, router_status, CoreType::ETH);
+                    }
+
+                    // Signal the kernel that it can start listening for packets
+                    tt_metal::detail::WriteToDeviceL1(
+                        dev, eth_logical_core, edm_config.edm_status_address, signal, CoreType::ETH);
+                }
+            }
+        };
+
+        for (const auto& dev : this->get_all_active_devices()) {
+            if (tt::Cluster::instance().get_associated_mmio_device(dev->id()) == dev->id()) {
+                auto tunnels_from_mmio = tt::Cluster::instance().get_tunnels_from_mmio_device(dev->id());
+                for (auto i = 0; i < tunnels_from_mmio.size(); i++) {
+                    // Need to poll on devices from farthest to the closest.
+                    for (auto j = tunnels_from_mmio[i].size() - 1; j > 0; j--) {
+                        wait_for_handshake(get_device(tunnels_from_mmio[i][j]));
+                    }
+                }
+                wait_for_handshake(dev);
+            }
+        }
```

Comment:
try to minimize the depth of the code to keep it more readable
```suggestion
        for (const auto& dev : this->get_all_active_devices()) {
            if (tt::Cluster::instance().get_associated_mmio_device(dev->id()) != dev->id()) {
                continue;
            }
            auto tunnels_from_mmio = tt::Cluster::instance().get_tunnels_from_mmio_device(dev->id());
            for (auto i = 0; i < tunnels_from_mmio.size(); i++) {
                // Need to poll on devices from farthest to the closest.
                for (auto j = tunnels_from_mmio[i].size() - 1; j > 0; j--) {
                    wait_for_handshake(get_device(tunnels_from_mmio[i][j]));
                }
            }
            wait_for_handshake(dev);
        }
```

Reply:
thanks! will update

Reply:
fixed